internal/processor/filterspan/config.go

```diff
@@ -108,3 +110,16 @@ type Attribute struct {
 	// If it is not set, any value will match.
 	Value interface{} `mapstructure:"value"`
 }
+
+// version match
+//  expected actual  match
+//  nil      <blank> yes
+//  nil      1       yes
+//  <blank>  <blank> yes
+//  <blank>  1       no
+//  1        <blank> no
+//  1        1       yes
+type InstrumentationLibrary struct {
+	Name    string  `mapstructure:"name"`
```

Comment:
I think it is the moment when we create a `StringMatcherConfig` struct:
```golang
type StringMatcherConfig struct {
  Type MatchType  // With a default maybe `strict`
  Value String
}
```

And use that here so users can decide for every string if it is a regexp or a strict match. What do you think? In a followup PR we may consider to update `SpanNames` and other strings to use the same struct.

We can also define an interface, and from every `StringMatcherConfig` we create an `StringMatcher` that is used in the matching logic:

```golang
type stringMatcher interface {
   match(input string) bool
}

type strinctStringMatcher interface {
  matchValue string
}

func (ssm *strinctStringMatcher) match(input string) bool {
  return matchValue == input
}

type regexpStringMatcher interface {
  matchPattern regexp.Pattern
}

func (rsm *regexpStringMatcher) match(input string) bool {
  return matchPattern.MatchString(input)
}

type alwaysTrueStringMatcher interface {
}

func (atsm *alwaysTrueStringMatcher) match(input string) bool {
  return true
}

type InstrumentationLibraryMatcher struct {
  nameMatcher StringMatcher
  verstionMatcher StringMatcher
}

func (ril *InstrumentationLibraryMatcher) Match(library pdata.InstrumentationLibrary) bool {
  return nameMatcher.match(library.Name()) && verstionMatcher.match(library.Version())
}
```

@tigrannajaryan @pjanotti any opinion on this?

Reply:
I like the idea.

Reply:
We should for the moment do this pattern only for Instrumentation library, then we can followup with changing the others.

Reply:
Let's wait few hours to see what @tigrannajaryan and @pjanotti also think :)

Reply:
no worries, I'm not going to do anything on in until Monday.

Reply:
@bogdandrutu since it's still unclear if implementation library will be removed, I make #929 a separate PR - but the idea of `StringMatcherConfig` would also make sense there. What do you think?